week7/answers.sql

```diff
@@ -1,3 +1,256 @@
+    FOREIGN KEY (character_id) REFERENCES characters(character_id),
+    FOREIGN KEY (item_id) REFERENCES items(item_id)
+);
+
+-- I'm trying to be as verbose as I can with JOINs
+-- Additionally, I like to indent like this so that only statements
+-- that are related to each other go in the same line
+CREATE VIEW character_items AS
+SELECT 
+    c.character_id,
+    c.name AS character_name, -- you told me to alias these a few weeks ago
+    i.name AS item_name,
+    i.armor,
+    i.damage
+FROM 
+    characters AS c
+INNER JOIN 
+    inventory AS inv ON c.character_id = inv.character_id -- not sure if this is correct
+INNER JOIN                                                -- UPDATE: It wasn't XD
+    items AS i ON inv.item_id = i.item_id
+UNION
+SELECT 
+    c.character_id,
+    c.name AS character_name,
+    i.name AS item_name,
+    i.armor,
+    i.damage
+FROM 
+    characters AS c
+INNER JOIN 
+    equipped AS e ON c.character_id = e.character_id
+INNER JOIN 
+    items AS i ON e.item_id = i.item_id -- this is where it should allow the view
+ORDER BY item_name;
+
+
+-- basically rinse repeat but for all of them in a team
+CREATE VIEW team_items AS
+SELECT 
+    t.team_id,
+    t.name AS team_name,
+    i.name AS item_name,
+    i.armor,
+    i.damage
+FROM 
+    teams t
+INNER JOIN 
+    team_members tm ON t.team_id = tm.team_id -- didn't know what else to call tm
+INNER JOIN 
+    characters c ON tm.character_id = c.character_id
+INNER JOIN 
+    inventory inv ON c.character_id = inv.character_id
+INNER JOIN 
+    items i ON inv.item_id = i.item_id
+UNION
+SELECT 
+    t.team_id,
+    t.name AS team_name,
+    i.name AS item_name,
+    i.armor,
+    i.damage
+FROM 
+    teams t
+INNER JOIN 
+    team_members tm ON t.team_id = tm.team_id
+INNER JOIN 
+    characters c ON tm.character_id = c.character_id
+INNER JOIN 
+    equipped e ON c.character_id = e.character_id
+INNER JOIN 
+    items i ON e.item_id = i.item_id;
+
+DELIMITER ;; -- I didn't know these delimeters could be set to be anything :-)
+CREATE FUNCTION armor_total(character_id INT) RETURNS INT
+DETERMINISTIC -- UPDATE - Github complained about this missing 
+BEGIN
+    DECLARE total_armor INT; -- I still struggle with when is it that vars need an @ symbol 
+			     -- before the declaration... I think they dont when they're local
+
+    SELECT SUM(i.armor) INTO total_armor
+    FROM equipped e
+    INNER JOIN items i ON e.item_id = i.item_id
+    WHERE e.character_id = character_id;
+
+    RETURN total_armor;
+END ;;
+DELIMITER ;
+
+DELIMITER ;; 
+CREATE PROCEDURE attack(IN id_of_character_being_attacked INT, IN id_of_equipped_item_used_for_attack INT)
+BEGIN
+    DECLARE armor INT;
+    DECLARE damage INT;
+    DECLARE health INT;
+
+    -- armor of character being attacked
+    SET armor = armor_total(id_of_character_being_attacked);
+
+    -- damage of item used to attack
+    SELECT damage INTO damage FROM items WHERE item_id = id_of_equipped_item_used_for_attack;
```

Comment:
This will give you the wrong damage. You are using the equipped_id to look something up by item_id. You need to figure out which item is associated with the equipped_id (you'll need a JOIN).